File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;
class Solution {
    
    public int solution(int[] people, int limit) {
       
        Arrays.sort(people);  //[40, 50, 80, 70]  40kg 이상이면 60넘는 사람은
        int count = 0;
        int start = 0;
        int end = people.length-1;
       // for(int i = 0; i< (int)Math.ceil(people.length/2); i++){
        while(start<=end) {
            
            if(people[end]+people[start]<=limit) {
  
            start++;
            count++;
            end--;
            }
            else{
               
                end--;
                count++;
            }
        }    
                    
        return count;
    }
}

File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {

    // 80 60 50 50 40 20 10
    let count = 0;
    let s = people.sort((a,b) => a-b);
    let left = 0;
    let right = s.length-1
    
    while(left <= right) {
        if(limit >=s[left] + s[right]) {
            left++;
        }
            right--;
            count++;
        
    }
    
    return count;
}